docroot/modules/custom/annotation_store/src/Controller/AnnotationStoreController.php

```diff
@@ -0,0 +1,192 @@
+<?php
+
+namespace Drupal\annotation_store\Controller;
+
+use Drupal\Component\Serialization\Json;
+use Symfony\Component\HttpFoundation\Request;
+
+/**
+ * Controller routines for annotation_store routes.
+ */
+class AnnotationStoreController {
+
+  /**
+   * Routing callback - annotation search.
+   */
+  public function annotationStoreSearch() {
+    $this->annotationReqType();
+  }
+
+  /**
+   * Routing callback - annotation save.
+   */
+  public function annotationStoreSave() {
+    $this->annotationReqType();
+  }
+
+  /**
+   * Routing callback - annotation update and delete.
+   */
+  public function annotationStoreUpdateDelete($id) {
+    $this->annotationReqType($id);
+  }
+
+  /**
+   * Get Resource Entity ID.
+   */
+  public function getResourceEntityId() {
+    $request_uri = \Drupal::request()->server->get('HTTP_REFERER');
```

Comment:
@anish-nirmal Wouldn't it be better here to just convert the path alias into normal Drupal path & take the nid from it?